stackslib/src/net/api/postmempoolquery.rs

```diff
@@ -276,10 +276,13 @@ impl RPCRequestHandler for RPCMempoolQueryRequestHandler {
         let page_id = self.page_id.take();
 
         let stream_res = node.with_node_state(|network, sortdb, chainstate, mempool, _rpc_args| {
-            let header = self.get_stacks_chain_tip(&preamble, sortdb, chainstate)
-                .map_err(|e| StacksHttpResponse::new_error(&preamble, &HttpServerError::new(format!("Failed to load chain tip: {:?}", &e))))?;
-
-            let coinbase_height = NakamotoChainState::get_coinbase_height(&mut chainstate.index_conn(), &header.index_block_hash())
+            let coinbase_height = NakamotoChainState::get_coinbase_height(
```

Comment:
Same here -- there's no reason this can't be cached within `network.stacks_tip`

Reply:
Addressed in [097c4db](https://github.com/stacks-network/stacks-core/pull/5308/commits/097c4db2e376f56794b0585347cb692585758bbe)